src/CoreFx.Private.TestUtilities/ref/CoreFx.Private.TestUtilities.cs

```diff
@@ -29,6 +29,7 @@ public static partial class AssertExtensions
         public static void ThrowsIf<T>(bool condition, System.Action action) where T : System.Exception { }
         public static T Throws<T>(System.Action action) where T : System.Exception { throw null; }
         public static void Throws<T>(System.Action action, string message) where T : System.Exception { }
+        public static TException Throws<TException, TResult>(Func<TResult> func) where TException : Exception { throw null; }
```

Comment:
Why is this needed, rather than just using `Assert.Throws`?

Reply:
I wanted to log the return value when an exception is not thrown